server/api/db/sqldb/db.py

```diff
@@ -2925,6 +2926,21 @@ def _update_feature_set_features(
                 )
                 update_labels(feature, labels)
                 feature_set.features.append(feature)
+            elif feature_name not in features_to_remove:
+                # check if the existing feature's labels were changed, and update them if needed
+                feature = next(
+                    (
+                        feature
+                        for feature in feature_set.features
+                        if feature.name == feature_name
+                    ),
+                    None,
+                )
```

Comment:
While this is cool, I got confused reading this, and assumed the comment of 'checking if the existing feature's labels were changed' was part of this generator with the filter.
I'd either change the comment to something like:
"If the feature already exists, and its labels were changed, update it with the new labels"
To explicitly express the steps you are taking in the code.

Reply:
Updated the comments